client/src/components/CreateNewBoardModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useContext, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import avatar from '../icons/avatar.jpg';
+import background from '../icons/teamBackground.jpg';
+import { DialogContext, ModalTypes } from '../contexts/DialogContext';
+
+const CreateNewBoardModal = () => {
+    const context = useContext(DialogContext);
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const onClickOutside = (e: any) => {
+        const element = e.target;
+        if (modalRef.current && !modalRef.current.contains(element)) {
+            e.preventDefault();
+            e.stopPropagation();
+            context.closeModalByType();
+        }
+    };
+
+    useEffect(() => {
+        document.body.addEventListener('click', onClickOutside);
+    });
```

Comment:
Whenever we make a subscription in a component, remember to **clean up all subscriptions** when the component unmounts. You can find out how to do it here: https://www.pluralsight.com/guides/how-to-cleanup-event-listeners-react